### DISCUSSION TITLE
Environment variables

### DISCUSSION THREAD

**Discussion by shiba-codes** [withastro/docs#5328] [Ideas]
Hi folks! Thanks for doing an awesome job 💟 

I stumbled upon a confusing behaviour yesterday which seems to be undocumented.
Imagine a scenario:
1. You have some secrets in gitlab that are exported as environment variables in CI
2. Your build script takes them and puts them into the `.env.production.local` file
3. Then, when all necessary variables are in place, `astro build` is called.

**What happened when built locally**
Variable values were interpolated into the built code like this:
`const API_TOKEN = "secret"`

**What happened in CI**
`const API_TOKEN = process.env.API_TOKEN`

After excessive googling and consulting the docs of both Astro and Vite we could not find anything that would explain this behaviour. Eventually we figured it out: when the variable is set but is also present in some .env file, it won't be set again. Instead, it will be expected to be present and be provided in runtime. Everything worked as expected after unsetting those variables.

I think this behaviour should be documented! What do you think?

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

Sorry no one could answer your question before but, Astro has now a [new way to handle environment variables](https://docs.astro.build/en/guides/environment-variables/#type-safe-environment-variables) and this behavior is now documented in https://docs.astro.build/en/guides/environment-variables/#variable-types. So, I'm going to close this as resolved.

If you think this is a mistake and that this discussion should remain open, please feel free to ping me, I'll take a second look at it.

Thank your for your contribution!
